src/AutoRest.CSharp.V3/Generation/Writers/ClientWriter.cs

```diff
@@ -331,6 +342,144 @@ private void WritePagingOperation(CodeWriter writer, Paging pagingMethod, bool a
+            Method originalMethod = lroMethod.OriginalMethod;
+            CSharpType responseType = new CSharpType(typeof(Operation<>), lroMethod.OriginalResponse.ResponseBody?.Type ?? new CSharpType(typeof(Response)));
+            Parameter[] parameters = lroMethod.CreateParameters;
+
+            writer.WriteXmlDocumentationSummary(originalMethod.Description);
+
+            foreach (Parameter parameter in parameters)
+            {
+                writer.WriteXmlDocumentationParameter(parameter.Name, parameter.Description);
+            }
+
+            writer.Append($"public {responseType} {CreateCreateOperationName(lroMethod.Name)}(");
+            foreach (Parameter parameter in parameters)
+            {
+                WriteParameter(writer, parameter);
+            }
+            writer.RemoveTrailingComma();
+            writer.Line($")");
+
+            using (writer.Scope())
+            {
+                WriteParameterNullChecks(writer, parameters);
+
+                writer.Append($"return {typeof(ArmOperationHelpers)}.Create(");
+                writer.Line($"pipeline, clientDiagnostics, originalResponse, {typeof(RequestMethod)}.{originalMethod.Request.HttpMethod.ToRequestMethodName()}, {originalMethod.Diagnostics.ScopeName:L}, {typeof(FinalStateVia)}.{lroMethod.FinalStateVia}, createOriginalHttpMessage,");
+
+                string valueVariable = "value";
+                const string document = "document";
+                ObjectSerialization? serialization = (lroMethod.OriginalResponse.ResponseBody as ObjectResponseBody)?.Serialization;
+                writer.Line($"(r, c) =>");
+                writer.LineRaw("{");
+                switch (serialization)
+                {
+                    case JsonSerialization jsonSerialization:
+                        writer.Append($"using var {document:D} = ");
+                        writer.Line($"{typeof(JsonDocument)}.Parse(r.ContentStream);");
+                        writer.ToDeserializeCall(
+                            jsonSerialization,
+                            w => w.Append($"document.RootElement"),
+                            ref valueVariable
+                        );
+                        writer.Line($"return {typeof(Response)}.FromValue({valueVariable}, r);");
+                        break;
+                    case XmlElementSerialization xmlSerialization:
+                        writer.Line($"var {document:D} = {typeof(XDocument)}.Load(r.ContentStream, LoadOptions.PreserveWhitespace);");
+                        writer.ToDeserializeCall(
+                            xmlSerialization,
+                            w => w.Append($"document"),
+                            ref valueVariable
+                        );
+                        writer.Line($"return {typeof(Response)}.FromValue({valueVariable}, r);");
+                        break;
+                    default:
+                        writer.Line($"return Response.FromValue(r, r);");
```

Comment:
Can we avoid passing lambdas at all if there is no return type?

Reply:
Technically, this is creating `Response<Response>`. That's because this lambda expects `Response<T>` to be returned. To avoid this, I'd have to complicate the logic to do UpdateStatus since it is what calls these lambdas. As it stands, either implementation isn't perfect because of this issue: https://github.com/Azure/autorest.csharp/issues/447
So, if we would need that logic, we can clean this up. Otherwise, the likelihood of this code actually being generated outside of the test server is almost zero.